src/canary.rs

```diff
@@ -174,7 +174,7 @@ fn round_up(n: u32, k: u32) -> u32 {
     if rem == 0 {
         n
     } else {
-        n + 4 - rem
+        n + k - rem
```

Comment:
Even though it's a private function, could you document it? Imho it's not immediately obvious that the function rounds up to next multiple of the given value `k`.